lib/options.js

```diff
@@ -41,3 +58,26 @@ export async function _initSecretKeySeed() {
 
   return {encoded: secretKeySeed, bytes: secretKeySeedBytes};
 }
+
+/**
+ * Parses a time-to-live value (in dayjs format), and converts it to a
+ * serialized expiration timestamp in XML DateTime format.
+ *
+ * @param {object} options - Options hashmap.
+ * @param {string} [options.ttl='1y'] - Time-to-live, in dayjs format.
+ *
+ * @returns {string} Serialized expiration timestamp.
+ */
+export function _parseTtl({ttl}) {
+  // split ttl into numbers and letters
+  ttl = ttl.match(/(\d+)(d|w|M|Q|y|h|ms|m|s)/);
```

Comment:
This looks fun. I don't want to think about it though -- hopefully someone else will review and double check this against whatever list is appropriate. If this comes from the internet somewhere, a link that would quickly explain it here as a comment would be great, i.e., `// valid time units are from <link>` and/or `// regex and valid time units are from <link>`. Also a bummer that `dayjs` can't just handle this bit ... maybe they have an "XML DateTime format" plugin we could depend on?

Reply:
@dlongley we did a similar implementation in veres-cli https://github.com/digitalbazaar/veres-cli/pull/11#discussion_r729321054
the units are from here: https://day.js.org/docs/en/durations/creating#list-of-all-available-units

Reply:
@dlongley  - I added a comment linking to the list of units above.